lib/authn/util/poll.ts

```diff
@@ -82,6 +83,48 @@ export function getPollFn(sdk, res: AuthnTransactionState, ref) {
       });
     }
 
+    const delayNextPoll = (ms) => {
+      // no need for extra logic in non-iOS environments, just continue polling
+      if (!isIOS()) {
+        return delayFn(ms);
+      }
+
+      let timeoutId: ReturnType<typeof setTimeout>;
+      const cancelableDelay = () => {
+        return new Promise((resolve) => {
+          timeoutId = setTimeout(resolve, ms);
+        });
+      };
+
+      const delayForFocus = () => {
+        let pageVisibilityHandler;
+        return new Promise<void>((resolve) => {
+          let pageDidHide = false;
+          pageVisibilityHandler = () => {
+            if (document.hidden) {
+              clearTimeout(timeoutId);
+              pageDidHide = true;
+            }
+            else if (pageDidHide) {
```

Comment:
Why is it important for us to track this `pageDidHide` state?

Reply:
It was an attempt to prevent changing behavior for authn polling flows when it isn't necessary. The code probably isn't necessary as the `Promise.race` would have already resolve if `pageDidHide` is `false` because the timeout wouldn't have been clear